packages/l10n/moment-utils.js

```diff
@@ -0,0 +1,113 @@
+      supportedLocales.some((supportedLocale) => {
+        const [language, countryCode] = supportedLocale.split('-');
+        if (countryCode) {
+          // Use an exact match.
+          return supportedLocale === locale;
+        }
+        // Fall back to the language tag.
+        return locale.startsWith(language);
+      })
+    )
+    // Keep it lowercase, as moment locale files are lowercase.
+    .map((locale) => locale.toLowerCase());
+
+  // Resolve the path to the moment locales folder.
+  const originalMomentLocalesPath = path.join(
+    path.dirname(require.resolve('moment')),
+    'locale'
+  );
+  // List all file names in the moment locales folder.
+  const momentLocaleDataFiles = fs
+    .readdirSync(originalMomentLocalesPath)
+    // Strip the extension part.
+    .map((fileName) => {
+      const [baseFileName] = fileName.split(path.extname(fileName));
+      return baseFileName;
+    });
+  // Assign to the available locales in the Merchant Center the corresponding
+  // matching moment locale, to make it easier to generate the code.
+  const allAvailableLocalesWithMatchingMomentLocale = allAvailableLocales
+    .map((availableLocale) => {
+      // Attempt to match a moment locale as-is.
+      const matchingLocale = momentLocaleDataFiles.find(
+        (momentLocale) => momentLocale === availableLocale
+      );
+      // Attempt to match a moment locale using only the language tag.
+      // This is used as a fallback in case the exact match didn't work.
+      const [language] = availableLocale.split('-');
+      const fallbackLocale = momentLocaleDataFiles.find(
+        (momentLocale) => momentLocale === language
+      );
+      const momentLocale = matchingLocale || fallbackLocale;
+      return {
+        locale: availableLocale,
+        momentLocale,
+      };
+    })
+    // Keep only locales that have a moment locale.
+    .filter(({ momentLocale }) => Boolean(momentLocale));
+
+  return allAvailableLocalesWithMatchingMomentLocale;
+}
+
+function getListOfSupportedMomentLocales() {
+  const allAvailableLocalesWithMatchingMomentLocale =
+    getListOfAvailableLocalesWithMatchingMomentLocale();
+
+  const supportedMomentLocales = allAvailableLocalesWithMatchingMomentLocale
+    .map(({ momentLocale }) => momentLocale)
+    // Filter out duplicates.
+    .filter(
+      (momentLocale, index, list) => list.indexOf(momentLocale) === index
+    );
+
+  return supportedMomentLocales;
+}
+
+function generateMomentLocaleImports() {
+  const allAvailableLocalesWithMatchingMomentLocale =
+    getListOfAvailableLocalesWithMatchingMomentLocale();
+
+  const code = `/* THIS IS A GENERATED FILE */
+
+async function loadMomentLocales(locale: string): Promise<void> {
+  const lowercaseLocale = locale.toLowerCase();
+
+  switch (lowercaseLocale) {
+  ${allAvailableLocalesWithMatchingMomentLocale
+    .map(
+      ({ locale, momentLocale }) => `
+    case '${locale}':
+      import('moment/locale/${momentLocale}');
```

Comment:
Since this is an `async` function, I think we would need to `await` for the `import` so we know for sure that the locale has been loaded after `loadMomentLocales` has been called.

```suggestion
      await import('moment/locale/${momentLocale}');
```

Reply:
Yes good point, thanks